### PR TITLE
Fix valgrind warning related to CStringDatum used instead of NameDatum plus some more fixes.

### PR DESCRIPTION
**Pull request by tanscorpio7**
### Description

This PR fixes multiple valgrind warnings.
1. Replaced all `CStringGetDatum` occurrences with `NameGetDatum` when doing CatalogTupleUpdate or Insert on a name column. NameDatum is always of NAMEDATALEN length and null padded if needed. 
2. Fix **string literal** to **rowversion** or **timestamp** cast warning. Row version is always of fixed length but when we cast string literal to rowversion for a smaller input string, we set the typmod correctly but do not actually blank pad the input string. As a fix we will simply reuse the varbinary/binary handling for string literals for row versions and timestamp, this is possible because row versions and timestamp were getting internally converted to varbinary type.
3. Make sure string is not empty before accessing its last character in `sp_execute_postgresql` & `sp_addrolemember` & `sp_droprolemember` &`sp_rename_internal`
4. In `TdsRecvTypeTable` we were not initialising all the entries of nulls array. Fixed that.
5. When logging TVP parameter we simply want to log the name, but we were sending CString as input with a record type which is wrong. To fix it, we will now use CString type for logging TVP name.

### Issues Resolved

[BABEL-5104]
[BABEL-5221]
[BABEL-4634]
[BABEL-4644]

### Test Scenarios Covered ###

Following JDBC tests do not throw any error with valgrind now
GRANT_SCHEMA
Test-sp_addrolemember-vu-prepar
Test-sp_addrolemember-vu-verify
Test-sp_addrolemember-vu-cleanu
babel-4475-vu-prepare
babel-4475-vu-verify
babel-4475-vu-cleanup
TestTvp -> with log_statements GUC set to 'all'

### Sign Off

Signed-off-by: Tanzeel Khan <tzlkhan@amazon.com>

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by shah-nirmit — APPROVED** [review]
LGTM , please check the tests failures..

File: contrib/babelfishpg_tsql/src/catalog.c [hunk 1]
```diff
@@ -3562,8 +3563,9 @@ rename_procfunc_update_bbf_catalog(RenameStmt *stmt)
 	initStringInfo(&new_funcsign);
 	appendStringInfoString(&new_funcsign, stmt->newname);
 	appendStringInfoString(&new_funcsign, strrchr(funcsign, '('));
+	namestrcpy(&newname_data, stmt->newname);
```
Comment (lejaokri) [inline contrib/babelfishpg_tsql/src/catalog.c:3566]: Where is this memory being freed?

Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/src/catalog.c:3566]: When `NameData newname_data` goes out of scope. We are copying the string into the static array in the NameData struct.

**Review by KushaalShroff — APPROVED** [review]

File: contrib/babelfishpg_tsql/src/catalog.c (hunk 1 above)
Comment (lejaokri) [reply contrib/babelfishpg_tsql/src/catalog.c:3566]: I missed this I thought namestrcpy would do a `palloc()`. This looks good to me

**Review by lejaokri — APPROVED** [review]

**Review by shalinilohia50 — APPROVED** [review]

**Pull request merged**